tests/auth_tests/test_admin_multidb.py

```diff
@@ -57,3 +61,16 @@ def test_add_view(self, mock):
                     },
                 )
                 mock.atomic.assert_called_with(using=db)
+
+    @mock.patch("django.contrib.auth.admin.transaction")
+    def test_read_only_methods_add_view(self, mock):
+        for db in self.databases:
+            for method in {"GET", "OPTIONS", "HEAD", "TRACE"}:
+                with self.subTest(db_connection=db, method=method):
+                    mock.mock_reset()
+                    Router.target_db = db
+                    self.client.force_login(self.superusers[db])
+                    self.client.request(
+                        url=reverse("test_adminsite:auth_user_add"), method=method
+                    )
```

Comment:
I don't think the test refactor is working as expected. If I revert the patch, these tests still pass as the response here is a 404

Reply:
Adding assertions to the other tests I added, the same is true there, too. I'll have a look into what's happening, and put some assertions on the `.status_code` in too for future.